googleworkspace/table_googleworkspace_admin_reports_activity.go

```diff
@@ -0,0 +1,216 @@
+        "calendar":                     true,
+        "chat":                         true,
+        "drive":                        true,
+        "gcp":                          true,
+        "gplus":                        true,
+        "groups":                       true,
+        "groups_enterprise":            true,
+        "jamboard":                     true,
+        "login":                        true,
+        "meet":                         true,
+        "mobile":                       true,
+        "rules":                        true,
+        "saml":                         true,
+        "token":                        true,
+        "user_accounts":                true,
+        "context_aware_access":         true,
+        "chrome":                       true,
+        "data_studio":                  true,
+        "keep":                         true,
+        "vault":                        true,
+        "gemini_in_workspace_apps":     true,
+}
+
+if !valid[appName] {
+    return nil, fmt.Errorf("unsupported application_name: %q", appName)
+}
+
+    // Build API call with the chosen category
+    call := service.Activities.List("all", appName)
+
+    if quals := d.Quals["time"]; quals != nil {
+        var startTime, endTime time.Time
+        for _, q := range quals.Quals {
+            if ts := q.Value.GetTimestampValue(); ts != nil {
+                t := ts.AsTime()
+                switch q.Operator {
+                case "=":
+                    startTime, endTime = t, t
+                case ">":
+                    startTime = t.Add(time.Nanosecond)
+                case ">=":
+                    startTime = t
+                case "<":
+                    endTime = t
+                case "<=":
+                    endTime = t
+                }
+            }
+        }
+        if !startTime.IsZero() {
+            call.StartTime(startTime.Format(time.RFC3339))
+        }
+        if !endTime.IsZero() {
+            call.EndTime(endTime.Format(time.RFC3339))
+        }
+    }
+
+    pageToken := ""
+    const apiMaxPageSize = 1000
+    var initialPageSize int64 = apiMaxPageSize
+    if limit := d.QueryContext.Limit; limit != nil && *limit < initialPageSize {
+        initialPageSize = *limit
+    }
+    call.MaxResults(initialPageSize)
+
+    for {
+        if pageToken != "" {
+            call.PageToken(pageToken)
+        }
+        resp, err := call.Do()
+        if err != nil {
+            plugin.Logger(ctx).Error("googleworkspace_admin_reports_activity.list", "api_error", err)
+            return nil, err
+        }
+        for _, activity := range resp.Items {
+            d.StreamListItem(ctx, activity)
+            if d.RowsRemaining(ctx) == 0 {
+                return nil, nil
+            }
+        }
+        if resp.NextPageToken == "" {
+            break
+        }
+        pageToken = resp.NextPageToken
+        if limit := d.QueryContext.Limit; limit != nil {

```

Comment:
Could you please update the pagination logic to align with the coding structure used in other tables? The goal is to maintain consistency across the codebase unless there's a strong reason to deviate from the standard approach.